src/test/java/net/snowflake/ingest/streaming/internal/datatypes/DateTimeIT.java

```diff
@@ -0,0 +1,1721 @@
+package net.snowflake.ingest.streaming.internal.datatypes;
+
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import org.junit.Ignore;
+import org.junit.Test;
+
+/**
+ * Supported date, time and timestamp formats:
+ * https://docs.snowflake.com/en/user-guide/date-time-input-output.html#date-formats
+ */
+public class DateTimeIT extends AbstractDataTypeTest {
+
+  @Test
+  public void testTimestampWithTimeZone() throws Exception {
+    useLosAngelesTimeZone();
+
+    // Test timestamp formats
+    testJdbcTypeCompatibility(
+        "TIMESTAMP_TZ",
```

Comment:
sorry for late comment but maybe for some future change
I was wondering if it makes sense to deduplicate the calls for tests with a lot of cases.
by declaring an array of input tuples, like:
```
cases = {
{"TIMESTAMP_TZ", "2/18/2008 02:36:48", "2008-02-18 02:36:48.000000000 -0800"},
...
};

for case in cases {
  testJdbcTypeCompatibility(case, ...);
}
```